mlflow/cli.py

```diff
@@ -519,17 +520,37 @@ def server(
     " are not specified, data is removed for all runs in the `deleted`"
     " lifecycle stage.",
 )
-def gc(older_than, backend_store_uri, run_ids):
+@click.option(
+    "--experiment-ids",
+    default=None,
+    help="Optional comma separated list of experiments to be permanently deleted including "
+    "all of their associated runs. If experiment ids are not specified, data is removed for all "
+    "experiments in the `deleted` lifecycle stage.",
+)
+def gc(older_than, backend_store_uri, run_ids, experiment_ids):
     """
     Permanently delete runs in the `deleted` lifecycle stage from the specified backend store.
     This command deletes all artifacts and metadata associated with the specified runs.
     """
+    import warnings
+    from mlflow.utils.time_utils import get_current_time_millis
+
     backend_store = _get_store(backend_store_uri, None)
+    skip_experiments = False
     if not hasattr(backend_store, "_hard_delete_run"):
         raise MlflowException(
             "This cli can only be used with a backend that allows hard-deleting runs"
         )
 
+    if not hasattr(backend_store, "_hard_delete_experiment"):
+        warnings.warn(
+            "This cli doesn't have the backend that allows hard-deleting experiments."
+            f" Update mflow version (current = {version}).",
```

Comment:
```suggestion
            f" Update mlflow version (current = {version}).",
```

Reply:
@shaikmoeed In this case, updating the MLflow version probably won't help (the maintainer of the third-party backend will need to implement support for hard deletion of experiments). Can we remove the "Update MLflow version" guidance?